src/main/java/com/example/onlinebookstore/repository/BookRepository.java

```diff
@@ -2,9 +2,12 @@
 
 import com.example.onlinebookstore.model.Book;
 import java.util.List;
+import java.util.Optional;
 
 public interface BookRepository {
     Book save(Book book);
 
     List<Book> findAll();
+
+    Optional<Book> getById(Long id);
```

Comment:
in repositories it's common to name methods `find`
```suggestion
    Optional<Book> findById(Long id);
```